src/features/metrics/aggregate/index.js

```diff
@@ -15,27 +15,28 @@ export class Aggregate extends AggregateBase {
   static featureName = FEATURE_NAME
   constructor (agentIdentifier, aggregator) {
     super(agentIdentifier, aggregator, FEATURE_NAME)
+    let scheduler
 
-    // Allow features external to the metrics feature to capture SMs and CMs through the event emitter
-    registerHandler(SUPPORTABILITY_METRIC_CHANNEL, this.storeSupportabilityMetrics.bind(this), this.featureName, this.ee)
-    registerHandler(CUSTOM_METRIC_CHANNEL, this.storeEventMetrics.bind(this), this.featureName, this.ee)
-
-    this.singleChecks() // checks that are run only one time, at script load
-    this.eachSessionChecks() // the start of every time user engages with page
-
-    var harvestTimeSeconds = getConfigurationValue(this.agentIdentifier, 'metrics.harvestTimeSeconds') || 30
-
-    var scheduler = new HarvestScheduler('jserrors', { onUnload: () => this.unload() }, this)
-    scheduler.harvest.on('jserrors', () => ({ body: this.aggregator.take(['cm', 'sm']) }))
-    this.ee.on(`drain-${this.featureName}`, () => { if (!this.blocked) scheduler.startTimer(harvestTimeSeconds) })
-
-    // if rum response determines that customer lacks entitlements for err endpoint, block it
+    // If RUM-call's response determines that customer lacks entitlements for the /jserror ingest endpoint, don't harvest at all.
     registerHandler('block-err', () => {
       this.blocked = true
-      scheduler.stopTimer()
+      if (scheduler) scheduler.aborted = true // RUM response may or may not have happened already before scheduler initialization below
     }, this.featureName, this.ee)
 
-    drain(this.agentIdentifier, this.featureName)
+    if (!this.blocked) {
```

Comment:
I'm not sure that there's really a point to wrap this with this if check.  The thing that blocks (`block-err`) will only trigger after draining (since its staged with registerHandler), which would always be after line 26 has already executed

Reply:
ok, figures. I thought `registerHandler` would be synchronous for some reason -- why isn't it actually? (Usually the handles are all in instrument and registers are all in aggs)

Reply:
Removed